mathesar/tests/views/api/test_table_api.py

```diff
@@ -873,3 +885,75 @@ def test_table_viewset_checks_cache(client):
     with patch.object(reflection, 'reflect_tables_from_schema') as mock_reflect:
         client.get('/api/v0/tables/')
     mock_reflect.assert_called()
+
+
+def test_table_retrieve_no_primary_key(create_table, client):
+    table_name = 'NASA Table Retrieve'
+    new_table_name = 'NASA Table Retrieve New'
+    table = create_table(table_name)
```

Comment:
You cannot use the existing `create_table` function here, since it creates a table with a primary key. You'll need to create a table without a primary key manually. Also applies to other `no_primary_key` functions below.

Reply:
How do I know from the table schema whether it will have zero, one or more than one primary keys?

Reply:
I'm not sure what you mean by the "table schema". You'll need to create the table, so you can create it with however many primary keys you like. 

Here's a tutorial on creating tables in SQLAlchemy: https://www.tutorialspoint.com/sqlalchemy/sqlalchemy_core_creating_table.htm. you can see that the `primary_key=True` argument is passed to a column. when creating the table for tests, you would instead either not pass that argument in for any column for testing no primary keys, or you would pass it to multiple columns to test composite primary keys.